lib/setupmanagers/qemuhck/qemuhck.rb

```diff
@@ -108,6 +111,33 @@ def initialize_clients_vm
       end
     end
 
+    def write_host_info
+      collect_vms_info
+      File.write("#{@workspace_path}/#{HOST_INFO_LOG_FILE}", compose_host_info)
+    end
+
+    def collect_vms_info
+      @vm_info = []
+      @vm_info << { name: 'studio', vnc_port: @studio_vm.vnc_port }
+      @platform['clients'].each do |(_k, v)|
+        @vm_info << { name: v['name'], vnc_port: @clients_vm[v['name']].vnc_port }
+      end
+    end
+
+    def compose_host_info
+      logs = []
+      logs << "Hostname: #{`hostname`.strip}"
+      logs << "QEMU version: #{`qemu-system-x86_64 --version`.lines.first.strip}"
+      logs << "Linux OS info: #{parse_os_release}"
+      logs << "Linux Kernel version: #{`uname -a`.strip}"
+      logs << "\nVMs Info:"
+
+      @vm_info.each do |vm|
+        logs << "VM Name: #{vm[:name]}, VNC Port: #{vm[:vnc_port]}"
```

Comment:
Do we also need an HMP\QMP port? It can be useful. Do we start QMP or HMP monitor?

Reply:
We start QMP, but it listens on a UNIX domain socket.

Reply:
Yes, we have HMP and QMP. Let's add it.

Reply:
@akihikodaki We still have HMP via telnet port.